test/test-cases/regression/variable-TX.json

```diff
@@ -80,5 +80,143 @@
       "SecRule REQUEST_HEADERS \"@rx ([A-z]+)\" \"id:1,log,pass,capture,id:14\"",
       "SecRule TX:0 \"@rx ([A-z]+)\" \"id:15\""
     ]
+  },
+  {
+    "enabled":1,
+    "version_min":300000,
+    "title":"Testing Variables :: capture group match after unused group",
+    "client":{
+      "ip":"200.249.12.31",
+      "port":123
+    },
+    "server":{
+      "ip":"200.249.12.31",
+      "port":80
+    },
+    "request":{
+      "uri":"/?key=aadd",
+      "method":"GET"
+    },
+    "response":{
+      "headers":{
+        "Date":"Mon, 13 Jul 2015 20:02:41 GMT",
+        "Last-Modified":"Sun, 26 Oct 2014 22:33:37 GMT",
+        "Content-Type":"text/html"
+      },
+      "body":[
+        "no need."
+      ]
+    },
+    "expected":{
+      "debug_log":"Added regex subexpression TX\\.3: dd[\\s\\S]*Target value: \"dd\" \\(Variable\\: TX\\:3[\\s\\S]*Rule returned 1"
+    },
+    "rules":[
+      "SecRuleEngine On",
+      "SecRule ARGS \"@rx (aa)(bb|cc)?(dd)\" \"id:1,log,pass,capture,id:16\"",
```

Comment:
Just a quick note: I assume this is just a typo, but surprising that a rule has two (different) `id`, and the engine allows that.

Also wanted to check the regression tests result, but there isn't any CI output. Did you merge this patch without that?

Reply:
Yes there should not be two instances of id.  I believe that happened because the preexisting tests in the file likewise had that, and I cut-and-pasted from there as a starting point.  Obviously an oversight on my part that I did not notice it.

Reply:
_and I cut-and-pasted from there as a starting point_ - sure, it's no problem, we're humans :).

But why the libmodsecurity3 allows this? And where it the regression test output?

Reply:
All the regression tests are available on **travis-ci**. There is a build for every commit. The history is available here: https://travis-ci.org/github/SpiderLabs/ModSecurity. That link is also available on the project description.

The checking is disabled on **v3/master** as a consequence of the development towards 3.1 as described here: 
[v3.1 Project](https://github.com/SpiderLabs/ModSecurity/projects/3)

Reply:
Thanks.